pkg/test/framework/components/istio/eastwest.go

```diff
@@ -64,18 +60,37 @@ func (i *operatorComponent) deployEastWestGateway(cluster resource.Cluster) erro
 		customEnv = append(customEnv, "SINGLE_CLUSTER=1")
 	}
 	cmd.Env = append(cmd.Env, customEnv...)
-	scopes.Framework.Infof("Deploying eastwestgateway in %s: %v %v", cluster.Name(), customEnv, generateSettings)
-	gwYaml, err := cmd.CombinedOutput()
+	gwIOP, err := cmd.CombinedOutput()
+	iopFile := path.Join(i.workDir, fmt.Sprintf("eastwest-%s.yaml", cluster.Name()))
+	err = ioutil.WriteFile(iopFile, gwIOP, os.ModePerm)
+	if err != nil {
+		return err
+	}
 
+	// use operator yaml to generate k8s resources
+	istioCtl, err := istioctl.New(i.ctx, istioctl.Config{Cluster: cluster})
 	if err != nil {
-		whichIstioctl, _ := exec.Command("which", "istioctl").CombinedOutput()
-		return fmt.Errorf("failed generating eastwestgateway manifest for %s using %q: %v: %s", cluster.Name(), whichIstioctl, err, string(gwYaml))
+		return err
 	}
-	i.saveManifestForCleanup(cluster.Name(), string(gwYaml))
-	// push the deployment to the cluster
-	if err := i.ctx.Config(cluster).ApplyYAML(i.settings.IngressNamespace, string(gwYaml)); err != nil {
+
+	genSettings := []string{
+		"manifest", "generate",
```

Comment:
should we use `install`? Does it work now? I think Martin fixed it

Reply:
Wouldn't that reconcile with the other deployed things and remove all but this gateway?

Reply:
if you add a name

```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  name: eastwest
```

it should work. As of some recent change. I tested it, seems to work ok.

If there are benefits to using generate for our own testing seems ok, but lets add the name so if someone does `gen.sh | istioctl install` it will not delete everything?

Reply:
No idea but things fail and don't give me an error message with `install`. I'll make a follow up because I do want to test that behavior.